### DISCUSSION TITLE
Outline running but blank - not using Cloudflare proxy or Cloudfront

### DISCUSSION THREAD

**Discussion by Zonalds** [outline/outline#2009] [Self Hosting]
I have outline running in an ec2 ubuntu 20.04 VM; image from the dockerhub. Not built from github. I am also running behind a proxy.

Everything checks out, at least it seems so from the logs. But the page is blank, only says Outline on the browser tab.

Here is the log:
```
2021-04-05T05:22:17.720Z services loaded backlinks service
2021-04-05T05:22:17.720Z services loaded debouncer service
2021-04-05T05:22:17.721Z services loaded importer service
2021-04-05T05:22:17.721Z services loaded notifications service
2021-04-05T05:22:17.721Z services loaded revisions service
2021-04-05T05:22:17.721Z services loaded slack service
2021-04-05T05:22:17.721Z services loaded websockets service
2021-04-05T05:22:17.882Z server loaded Email auth provider
2021-04-05T05:22:17.883Z server loaded Google auth provider
2021-04-05T05:22:17.883Z server loaded Slack auth provider

> Listening on http://localhost:3000
```

I have port 3000 exposed to my vm via `-p 5000:3000` as the port in the log isn't exposed. 

My URL and port configurations in the .env file are as follows:

```
URL=http://localhost:3000
PORT=3000
```
Then I exposed container as a directory from my nginx (which has other services like blogs and main site running on it) using:

```
 location  ^~ /guide {
        proxy_pass http://localhost:5000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
```

I also tried:

```
    location ^~ /guide {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $http_host;
        proxy_pass http://localhost:5000;
        proxy_redirect off;
    }
```

On trying to access outline at https://mydomain.com/guide, the page would return blank.

Any idea what I could be doing wrong?

**Comment by tommoor** [comment]
Sounds like you maybe need to configure Slack or Google auth in the .env file… ?

**Reply by Zonalds** [reply]
Hi, I already did. I added both.

I would notice that the html(header and script) loaded when I explore page source. can you take a look at mutlist.com/guide

**Reply by Zonalds** [reply]
This seems to only happen when I try to run it as a sub-directory e.g https://mydomain.com/guide. But runs well on a subdomain. Am I doing ti wrong?

**Reply by tommoor** [reply]
Running as a subdirectory is not supported, there are a _lot_ of assumptions in the codebase that the project has it's own domain.
